src/index.ts

```diff
@@ -19,39 +26,88 @@ interface Trie {
   put(key: Buffer | string, value: Buffer | string, cb: TriePutCb): void
 }
 
+interface AccountProps {
+  nonce: Buffer
+  balance: Buffer
+  stateRoot: Buffer
+  codeHash: Buffer
+}
+
+// Would prefer for Account class getters to have type `Buffer`
+// and setters to have type `any` to be cast `toBuffer`,
+// however get ts error:
+// `'get' and 'set' accessor must have the same type`
+type BufferLike = Buffer | any
+
 export default class Account {
+  private props: AccountProps
+
   /**
    * The account's nonce.
    */
-  public nonce!: Buffer
+  get nonce(): BufferLike {
+    return this.props.nonce
+  }
+
+  set nonce(nonce: BufferLike) {
+    this.props.nonce = toBuffer(nonce)
```

Comment:
These things always confuse me, but isn't the nonce a number? As per RLP it should have it's leftmost zeros stripped.

Reply:
On a second thought, account representations are never transmitted, nor used in consensus rules, so I guess we are free to represent them as we want.